iosApp/SurveyTests/Sources/Specs/Presentation/Modules/Login/LoginViewDataSourceSpec.swift

```diff
@@ -0,0 +1,139 @@
+//  swiftlint:disable function_body_length
+//
+//  LoginViewDataSourceSpec.swift
+//  Survey
+//
+//  Created by Bliss on 28/10/22.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Nimble
+import Quick
+import Shared
+
+@testable import Survey
+
+final class LoginViewDataSourceSpec: QuickSpec {
+
+    override func spec() {
+
+        var loginCoordinator: LoginCoordinatorMock!
+        var loginUseCase: LogInUseCaseKMMMock!
+        var loginViewModel: LoginViewModel!
+        var dataSource: LoginView.DataSource!
+
+        describe("a Login View Data Source") {
+
+            beforeEach {
+                loginCoordinator = LoginCoordinatorMock()
+                loginUseCase = LogInUseCaseKMMMock()
+                loginViewModel = LoginViewModel(logInUseCase: loginUseCase)
+                dataSource = .init(
+                    coordinator: loginCoordinator,
+                    viewModel: loginViewModel
+                )
+            }
+
+            describe("its init") {
```

Comment:
Could you clarify this, this is describing the function of the class so it make sense it use `describe("its`?

Reply:
ohh so sorry if I create any confusion for you, I mean we should use `context` with when instead of `describe` for function as from Compass example we are doing that 🙏 

![Screenshot 2022-12-16 at 15 03 41](https://user-images.githubusercontent.com/70877098/208051994-1190325a-514a-4e4a-a546-4e9420bbb70c.png)

This is also the practice our team is doing, for example the unit tests in the MVVMRxSwiftDemo project 😁

https://github.com/nimblehq/mvvm-rxswift-demo/blob/30c9a010378b45509a540a329c943ffbec27048a/MVVMRxSwiftDemoTests/Sources/Specs/Presentation/Modules/Login/LoginViewControllerSpec.swift#L1-L74

Let me know if you have any confusion 🙇‍♂️

Reply:
From my understanding from this [conversation](https://nimble-co.slack.com/archives/C418DBL2W/p1660810014305399) and [this discussion](https://github.com/nimblehq/compass/discussions/1048), we are using `describe` to define what the class can do (func), we use `context` for giving more logic to the action "when". So I think it's best to keep the format as we discussed throughout the chapter?

Reply:
@blyscuit oh you are right, I totally forget about our discussions and I was just checking the current content on Compass, I think we are safe to ignore my comment here, nice recall 👍